packages/flow-core/src/components/f-grid/f-grid.ts

```diff
@@ -0,0 +1,64 @@
+import { html, unsafeCSS } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import eleStyle from "./f-grid.scss";
+import { FRoot } from "../../mixins/components/f-root/f-root";
+import { FDiv } from "../f-div/f-div";
+
+export type FGridBodyHeightProp = `${number}px` | `${number}%` | `${number}vh`;
+
+@customElement("f-grid")
+export class FGrid extends FRoot {
+	/**
+	 * css loaded from scss file
+	 */
+	static styles = [unsafeCSS(eleStyle), ...FDiv.styles];
+
+	/**
+	 * @attribute Defines the minimum value of width of each cell. A cell can not be smaller than this in any viewport.
+	 */
+	@property({ reflect: true, type: Number, attribute: "min-cell-width" })
+	minCellWidth?: number;
+
+	/**
+	 * @attribute Defines the maximum width of each cell.
+	 */
+	@property({ reflect: true, type: Number, attribute: "max-cell-width" })
+	maxCellWidth?: number;
+
+	/**
+	 * @attribute Defines the height of each cell
+	 */
+	@property({ reflect: true, type: Number, attribute: "cell-height" })
+	cellHeight?: number;
+
+	/**
+	 * @attribute Gap defines the space between the items of a f-grid
+	 */
+	@property({ type: String, reflect: true })
+	gap?: "x-large" | "large" | "medium" | "small" | "x-small" = "small";
+
+	/**
+	 * styling for grid
+	 */
+	applyGridStyles() {
+		this.style.gridTemplateColumns = `repeat(auto-fill,minmax(${
```

Comment:
the ternary operator is good for minor conditions. Can you create `get` property for the right side of an assignment